examples/functional_test.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python3
+"""Functional test script for image classification using OAuth authentication.
+
+This test is primarily used internally, but is provided here as a further
+example of using the Athena client with OAuth.
+"""
+
+import asyncio
+import logging
+import os
+import sys
+import time
+import uuid
+from collections.abc import AsyncIterator
+
+from create_image import iter_images
+from dotenv import load_dotenv
+
+from resolver_athena_client.client.athena_client import AthenaClient
+from resolver_athena_client.client.athena_options import AthenaOptions
+from resolver_athena_client.client.channel import (
+    CredentialHelper,
+    create_channel_with_credentials,
+)
+from resolver_athena_client.client.models.input_model import ImageData
+from resolver_athena_client.client.utils import (
+    get_output_error_summary,
+    has_output_errors,
+    process_classification_outputs,
+)
+from resolver_athena_client.generated.athena.models_pb2 import (
+    ClassificationOutput,
+    ClassifyResponse,
+)
+
+
+async def rate_limited_image_iter(
+    min_interval_ms: int,
+    max_images: int | None = None,
+    counter: list[int] | None = None,
+) -> AsyncIterator[ImageData]:
+    """Generate images with a minimum interval between yields."""
+    last_yield_time = time.time()
+    async for image in iter_images(max_images, counter):
+        elapsed_ms = (time.time() - last_yield_time) * 1000
+        if elapsed_ms < min_interval_ms:
+            await asyncio.sleep((min_interval_ms - elapsed_ms) / 1000)
+        yield image
+        last_yield_time = time.time()
+
+
+def process_errors(
+    logger: logging.Logger, result: ClassifyResponse, current_error_count: int
+) -> int:
+    """Process errors in the result and update the current error count."""
+    if has_output_errors(result):
+        error_summary = get_output_error_summary(result)
+        result_error_count = sum(error_summary.values())
```

Comment:
I wonder if `error_counts` is a better name for this. probably that method is just bad naming. should ge `get_output_error_counts` or smth. minor anyways.